apps/rtc/lib/rtc_web/channels/shell_channel.ex

```diff
@@ -7,20 +7,15 @@ defmodule RtcWeb.ShellChannel do
   require Logger
 
   def join("shells:me", _, socket) do
-    send(self(), :connect)
-    {:ok, socket}
-  end
-
-  def handle_info(:connect, socket) do
     with %CloudShell{pod_name: name} = shell <- Shell.get_shell(socket.assigns.user.id),
          {:ok, _} <- Client.setup(shell),
          url <- Pods.PodExec.exec_url(name),
          {:ok, pid} <- Pods.PodExec.start_link(url, self()) do
-      {:noreply, assign(socket, :wss_pid, pid)}
+      {:ok, assign(socket, :wss_pid, pid)}
     else
       err ->
         Logger.info "failed to exec pod with #{inspect(err)}"
-        {:stop, {:shutdown, :failed_exec}, socket}
+        {:error, err}
```

Comment:
```elixir
Process.send_after(self(), :shutdown, 10 * 1000)
push(...)
{:ok, socket}
```

then add another handle_info like

```elixir
def handle_info(:shutdown, socket), do: {:stop, {:shutdown, :failed_boot}, socket}
```

Reply:
actually thinking about this a bit more, the scale of our shell processes are so low, this optimization shouldn't really be necessary